packages/ciphernode/net/src/dialer.rs

```diff
@@ -1,210 +1,261 @@
 use libp2p::{
     multiaddr::Protocol,
     swarm::{dial_opts::DialOpts, ConnectionId, DialError},
     Multiaddr,
 };
-use std::net::ToSocketAddrs;
-use tokio::select;
-use tokio::sync::{broadcast, mpsc};
-use tokio::time::{sleep, Duration};
-use tracing::error;
-use tracing::info;
-
-use crate::{
-    events::{NetworkPeerCommand, NetworkPeerEvent},
-    retry::{retry_with_backoff, to_retry, RetryError, BACKOFF_DELAY, BACKOFF_MAX_RETRIES},
-};
+use std::{collections::HashMap, net::ToSocketAddrs, sync::Arc, time::Duration};
+use tokio::sync::mpsc;
+use tracing::{info, warn};
 
-/// Dial a single Multiaddr with retries and return an error should those retries not work
-async fn dial_multiaddr(
-    cmd_tx: &mpsc::Sender<NetworkPeerCommand>,
-    event_tx: &broadcast::Sender<NetworkPeerEvent>,
-    multiaddr_str: &str,
-) -> Result<()> {
-    let multiaddr = &multiaddr_str.parse()?;
-    info!("Now dialing in to {}", multiaddr);
-    retry_with_backoff(
-        || attempt_connection(cmd_tx, event_tx, multiaddr),
-        BACKOFF_MAX_RETRIES,
-        BACKOFF_DELAY,
-    )
-    .await?;
-    Ok(())
-}
+const BACKOFF_DELAY: u64 = 500;
+const BACKOFF_MAX_RETRIES: u32 = 10;
+const CONNECTION_TIMEOUT: u64 = 60;
 
-fn trace_error(r: Result<()>) {
-    if let Err(err) = r {
-        error!("{}", err);
-    }
-}
+#[derive(Message)]
+#[rtype(result = "Result<()>")]
+pub struct DialPeer(pub String);
 
-/// Initiates connections to multiple network peers
-///
-/// # Arguments
-/// * `cmd_tx` - Sender for network peer commands
-/// * `event_tx` - Broadcast sender for peer events
-/// * `peers` - List of peer addresses to connect to
-pub async fn dial_peers(
-    cmd_tx: &mpsc::Sender<NetworkPeerCommand>,
-    event_tx: &broadcast::Sender<NetworkPeerEvent>,
-    peers: &Vec<String>,
-) -> Result<()> {
-    let futures: Vec<_> = peers
-        .iter()
-        .map(|addr| dial_multiaddr(cmd_tx, event_tx, addr))
-        .collect();
-    let results = join_all(futures).await;
-    results.into_iter().for_each(trace_error);
-    Ok(())
+#[derive(Clone)]
+struct PendingConnection {
+    addr: String,
+    attempt: u32,
+    delay_ms: u64,
 }
 
-/// Attempt a connection with retrys to a multiaddr return an error if the connection could not be resolved after the retries.
-async fn attempt_connection(
-    cmd_tx: &mpsc::Sender<NetworkPeerCommand>,
-    event_tx: &broadcast::Sender<NetworkPeerEvent>,
-    multiaddr: &Multiaddr,
-) -> Result<(), RetryError> {
-    let mut event_rx = event_tx.subscribe();
-    let multi = get_resolved_multiaddr(multiaddr).map_err(to_retry)?;
-    let opts: DialOpts = multi.clone().into();
-    let dial_connection = opts.connection_id();
-    info!("Dialing: '{}' with connection '{}'", multi, dial_connection);
-    cmd_tx
-        .send(NetworkPeerCommand::Dial(opts))
-        .await
-        .map_err(to_retry)?;
-    wait_for_connection(&mut event_rx, dial_connection).await
+#[derive(Clone)]
+pub struct DialerActor {
```

Comment:
Could call this `Dialer`? just we haven't used the `Actor` prefix before.

Reply:
Yup!